eksupgrade/models/eks.py

```diff
@@ -0,0 +1,357 @@
+            clusterName=self.cluster.name,
+            addonName=self.name,
+            addonVersion=version,
+            resolveConflicts="OVERWRITE",
+            **self._addon_update_kwargs,
+        )
+        return update_response
+
+
+@dataclass
+class Cluster(EksResource):
+    """Define the Kubernetes Cluster model."""
+
+    certificate_authority_data: str = ""
+    identity_oidc_issuer: str = ""
+    endpoint: str = ""
+    role_arn: str = ""
+    platform_version: str = ""
+    secrets_key_arn: str = ""
+    cluster_logging_enabled: bool = False
+
+    def __repr__(self) -> str:  # pragma: no cover
+        """Return the string representation of a Cluster."""
+        return f"<{self.__class__.__name__} - Name: {self.name} | Version: {self.version} | Last Status: {self.status}>"
+
+    def __post_init__(self) -> None:
+        """Perform the post initialization steps."""
+        self._register_k8s_aws_id_handlers()
+        self.load_config()
+
+    def _register_k8s_aws_id_handlers(self) -> None:
+        """Register the kubernetes AWS ID header handlers."""
+        self.sts_client.meta.events.register(
+            "provide-client-params.sts.GetCallerIdentity",
+            self._retrieve_k8s_aws_id,
+        )
+        self.sts_client.meta.events.register(
+            "before-sign.sts.GetCallerIdentity",
+            self._inject_k8s_aws_id_header,
+        )
+
+    def _retrieve_k8s_aws_id(self, params, context, **_) -> None:
+        """Retrieve the kubernetes AWS ID header for use in boto3 request headers."""
+        if TOKEN_HEADER_KEY in params:
+            context[TOKEN_HEADER_KEY] = params.pop(TOKEN_HEADER_KEY)
+            logger.debug("Retrieving cluster header %s: %s", TOKEN_HEADER_KEY, context[TOKEN_HEADER_KEY])
+
+    def _inject_k8s_aws_id_header(self, request, **_) -> None:
+        """Inject the kubernetes AWS ID header into boto3 request headers."""
+        if TOKEN_HEADER_KEY in request.context:
+            request.headers[TOKEN_HEADER_KEY] = request.context[TOKEN_HEADER_KEY]
+            logger.debug("Patching boto3 STS calls with cluster headers: %s", request.headers)
+
+    def _get_presigned_url(self, url_timeout: int = 60) -> str:
+        """Get the presigned URL."""
+        logger.debug("Generating the pre-signed url for get-caller-identity...")
+        return self.sts_client.generate_presigned_url(
+            "get_caller_identity",
+            Params={TOKEN_HEADER_KEY: self.cluster_identifier},
+            ExpiresIn=url_timeout,
+            HttpMethod="GET",
+        )
+
+    @cached_property
+    def _current_addons(self) -> List[str]:
+        """Return a list of addon names currently installed in the cluster."""
+        logger.debug("Getting the list of current cluster addons for cluster: %s...", self.name)
+        return self.eks_client.list_addons(clusterName=self.name).get("addons", [])
+
+    @property
+    def cluster_name(self) -> str:
+        """Return the cluster ID."""
```

Comment:
Cluster ID is something different (as you have below)
```suggestion
        """Return the cluster name."""
```